### ISSUE TITLE
NewtonNonlinearSolver returns nan when initialized with root value

### ISSUE THREAD

**Issue by fhchl** [patrick-kidger/diffrax#296]
It seems like the Newton solver fails when initialized with the root value at which also $f'(x) = 0$. A minimal working example:

```python
from diffrax import NewtonNonlinearSolver

def fun(x, args): return x**3
solver = NewtonNonlinearSolver(rtol=1e-3, atol=1e-6)
sol = solver(fun, 0., None)
print(sol.root, fun(sol.root, None))  # nan nan
```

It looks like the conditions [here](https://github.com/patrick-kidger/diffrax/blob/104936ab1c5861235f766bdf3256018525670213/diffrax/nonlinear_solver/newton.py#L128C1-L128C89) would catch it, where it not for `at_least_two` iterations.

**Comment by patrick-kidger** [comment]
Yeah, this is due to the linear solve `Ax=b` basically turning into `0 @ x = 0`, which isn't defined.
The current nonlinear code is due to get overhauled fairly shortly! I'll mark this as a bug until then.
